File: src/main/java/com/uapp_llc/service/TaskService.java
package com.uapp_llc.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.uapp_llc.model.Column;
import com.uapp_llc.model.Task;

public interface TaskService {

  Task create(Column column, String name, String description);

  Task update(Long id, Long columnId, String name, String description);

  Task move(Long id, Long columnId, Column newColumn, Integer newIndex);

  Task find(Long id, Long columnId);

  Page<Task> findAll(Long columnId, Pageable pageable);

  void delete(Long id, Long columnId);

}

File: src/main/kotlin/com/uapp_llc/service/TaskService.kt
package com.uapp_llc.service

import com.uapp_llc.model.Column
import com.uapp_llc.model.Task
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface TaskService {

  fun create(column: Column, name: String, description: String): Task

  fun update(id: Long, columnId: Long, name: String?, description: String?): Task

  fun move(id: Long, columnId: Long, newColumn: Column?, newIndex: Int?): Task

  fun find(id: Long, columnId: Long): Task

  fun findAll(columnId: Long, pageable: Pageable): Page<Task>

  fun delete(id: Long, columnId: Long)

}
